torchxrayvision/datasets.py

```diff
@@ -195,7 +197,24 @@ def __len__(self):
     def __getitem__(self, idx):
         return self.dataset[self.idxs[idx]]
 
-    
+
+class TarDataset(Dataset):
+    def __init__(self, imgpath):
+        if imgpath.endswith(".tar"):
```

Comment:
What about using tarfile.is_tarfile ? This will allow for compressed files which people may want to use.

Reply:
Good idea, thanks